synapse/federation/transport/server.py

```diff
@@ -396,6 +407,24 @@ def on_POST(self, origin, content, query, room_id):
         defer.returnValue((200, content))
 
 
+class On3pidBindServlet(BaseFederationServlet):
+    PATH = "/3pid/onbind"
+
+    @defer.inlineCallbacks
+    def on_POST(self, request):
+        content_bytes = request.content.read()
+        content = json.loads(content_bytes)
+        if "invites" in content:
+            for invite in content["invites"]:
+                yield self.handler.exchange_third_party_invite(invite)
+        defer.returnValue((200, {}))
```

Comment:
Personally, I would do the the processing of the content in the `federation_server`, this vaguely preserving the vague separation of parsing the request and handling the request?